Add mention of Composer and PHP compatibility to project README.

Potherca opened the pull request:
Fixes #129

jrfnl commented on README.md:
```diff
@@ -29,6 +27,14 @@ composer require --dev dealerdirect/phpcodesniffer-composer-installer
+
+PHP **5.x**, **7.x**, and **8.x** are all supported.
+
+For PHP 8 version `0.7.x` (or higher) of this plugin must be _explicitly_ set as  version constraints, as Composer treats minors releases below 1.0 as major releases. In other words: using `^0.6` will not allow Composer to install v0.7 and thus _not_ support PHP8.
```
This goes for _both_ Composer 2.x as well as PHP 8.0.

Suggestion for alternative content of this section:

> This plugin is compatible with:
> * PHP 5.x, 7.x and 8.x.
>     Support for PHP 8.0 was added in `v0.7.0`.
> * [Composer][composer] 1.x and 2.x.
>     Support for Composer 2.x was added in `v0.7.0`.
> * [PHP_CodeSniffer][codesniffer] 2.x and 3.x.
>     Support for PHP_CodeSniffer 3.x was added in `v0.4.0`.
>
> Note: as Composer treats minor releases below 1.0.0 as major releases, version `0.7.x` (or higher) of this plugin must be _explicitly_ set as version constraint when using Composer 2.x or PHP 8.0.
> In other words: using `^0.6` will not work with Composer 2.x or PHP 8.0.

Potherca replied:
Will amend.

Potherca commented:
@jrfnl  I finally found some time to update this! :tada: 

I've changed the wording, as you suggested. Some things to note:

- I added links to various things, just in case people want to know more
- I placed the comment in a block (using `>`) to make it similar to the warning block further down in the document
- I added an emoji as icon for the block, to make it stand out more (and look similar to the other block). I didn't feel this message warranted a "warning", so I opted for "Information" instead.

jrfnl commented on README.md:
```diff
@@ -29,6 +27,17 @@ composer require --dev dealerdirect/phpcodesniffer-composer-installer
+
+This plugin is compatible with:
+
+- PHP **5.x**, **7.x**, and **8.x** (Support for PHP v8 is available since [`v0.7.0`][v0.7])
```
Just placing a note here for the record:
While we currently can only guarantee support with PHP 8.0, the current test build against PHP 8.1 _does_ seem to pass, so I'm fine with leaving this at `v8`.
